src/Stratis.SmartContracts.CLR/Loader/ContractAssemblyLoader.cs

```diff
@@ -1,27 +1,127 @@
 ﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.IO.Compression;
+using System.Net;
 using System.Reflection;
+using System.Runtime.Loader;
 using CSharpFunctionalExtensions;
+using Microsoft.Extensions.Logging;
 
 namespace Stratis.SmartContracts.CLR.Loader
 {
+    public class SmartContractLoadContext : AssemblyLoadContext
+    {
+        private readonly ILogger logger;
+        private readonly AssemblyLoadContext defaultContext;
+        private static Dictionary<string, byte[]> cache = new Dictionary<string, byte[]>();
+
+        public SmartContractLoadContext(AssemblyLoadContext defaultContext, ILogger logger)
+        {
+            this.logger = logger;
+            this.defaultContext = defaultContext;
+        }
+
+        public static string GetExactAssembly(AssemblyName assemblyName, out string errorMessage)
+        {
+            // The DLL is not included with our distribution. See if we can get it from NuGet.org.
+            string version = assemblyName.Version.ToString();
+            string folderName = $"{assemblyName.Name.ToLower()}.{version}";
+            string assemblyFolder = Path.GetFullPath("LegacyStandardsDLLs");
+            string assemblyPath = Path.Combine(assemblyFolder, folderName);
+
+            errorMessage = null;
+
+            if (!Directory.Exists(assemblyPath))
+            {
+                string downloadFile = $"{assemblyPath}.nupkg";
+
+                if (!File.Exists(downloadFile))
+                {
+                    Directory.CreateDirectory(assemblyFolder);
+
+                    string downloadLink = $"https://www.nuget.org/api/v2/package/{assemblyName.Name.ToLower()}/{version}";
```

Comment:
Downloading assembly packages from a central resource is not safe and it is against single point of failure and also it is a security issue.

Reply:
Then I suggest that we simply pop the message and have the user do this manually. WDYT?

Reply:
I've added some code to limit what can be done in the standards package - similar to what we do for the smart contracts.

Reply:
Since there is an approval process involved with smart contracts another approach would be to release the upgraded standards package and simply not approve new contracts based on it until a sufficient number of node have upgraded. This way the nodes don't have to pull any package on demand and can simply use the appropriate package from the LegacyStandardsDLLs folder.